packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -3908,6 +3908,47 @@ export class ProjectService extends BaseService {
         return spacesWithUserAccess;
     }
 
+    async createPreview(
+        user: SessionUser,
+        projectUuid: string,
+        data: {
+            name: string;
+            copyContent: boolean;
+        },
+        context: RequestMethod,
+    ): Promise<string> {
+        // create preview project permissions are checked in `createWithoutCompile`
+        const project = await this.projectModel.getWithSensitiveFields(
+            projectUuid,
+        );
+
+        if (!project.warehouseConnection) {
+            throw new ParameterError(
+                `Missing warehouse connection for project ${projectUuid}`,
+            );
+        }
+        const previewData: CreateProject = {
+            name: data.name,
+            type: ProjectType.PREVIEW,
+            warehouseConnection: project.warehouseConnection,
+            dbtConnection: project.dbtConnection,
+            upstreamProjectUuid: data.copyContent ? projectUuid : undefined,
+            dbtVersion: project.dbtVersion,
+        };
+
+        const previewProject = await this.createWithoutCompile(
+            user,
+            previewData,
+            context,
+        );
+        await this.scheduleCompileProject(
```

Comment:
since this is scheduled, do we want to be more rigorous with the loading state of this request on the FE?

Reply:
maybe, by the time the user goes into an explore in preview, hopefully the project has compiled. If not, we can consider copying the existing explores from one project to another. I'd like to try this first as it is and get some feedback.